### PR TITLE
integrate db object

### PR DESCRIPTION
**Pull request by tohuynh**
Rewrote /api files to use new db object.

Issue 8: Sort events by event_datetime in /events

I also changed date formats to ISO format and then always display date using local format.

> Also as general feedback, this is not an issue for now, but in the future, it may be useful to revisit how sorting is currently being done client side. GH #24 and the related branch has new database interaction functions that allow for sorting on server side which should reduce render times.

I added sorting on server side where I could, but render time didn't improve much overall.
I think the slow render time, for example in /people/#person_id, has more to do with firebase. In personApi.js, line 27-30, that's where most of slow time comes from.  It is suppose to get all event_minutes_item concurrently, but it seems like firebase isn't able to handle all the requests for events_minutes_item simultaneously.

### CODE REVIEW

File: src/api/database.js [hunk 1]
```diff
@@ -4,16 +4,16 @@ import { isString } from "./utils";
 
 export const WHERE_OPERATORS = {
     eq: "==",
-    contains: "in",
+    contains: "array-contains",
```
Comment (evamaxfield) [inline src/api/database.js:7]: Really nice catch. Just double checked docs.

File: src/api/database.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    * @return {Object[]} The formatted query results.
+    */
+    async selectRowsAsArray(table, filters = [], orderBy = undefined, limit = 1000) {
+        try {
+            // Make basic reference
+            let ref = this.connection.collection(table);
+
+            // Attach filters
+            filters.forEach(filter => {
+                ref = ref.where(filter.columnName, filter.operator, filter.value);
```
Comment (evamaxfield) [inline src/api/database.js:215]: From commit history, nice catch here as well.

File: src/api/eventApi.js [hunk 3]
```diff
@@ -2,69 +2,53 @@ import * as firebase from "firebase";
...
 import { flatten, sortBy, reverse, groupBy, mapValues, filter } from "lodash";
-import { getAllResources, getSingleResource, getResourceById } from "./baseApi";
+import { db, WhereCondition, WHERE_OPERATORS, OrderCondition, ORDER_OPERATORS } from "./database";
 import { fetchJson } from "./utils";
 
 const storage = firebase.storage();
 
 export async function getAllEvents() {
-  return getAllResources("event");
+  return await db.selectRowsAsArray("event", [], new OrderCondition(["event_datetime", ORDER_OPERATORS.desc]));
```
Comment (evamaxfield) [inline src/api/eventApi.js:11]: I think this also closes #8.

File: src/components/EventCard.jsx [hunk 4]
```diff
@@ -17,7 +18,7 @@ const EventCard = ({ name, date, description, link }) => (
       <Card.Header>
         <Link to={link}>{name}</Link>
       </Card.Header>
-      <Card.Meta>{date}</Card.Meta>
+      <Card.Meta>{moment.utc(date).format("LLL")}</Card.Meta>
```
Comment (evamaxfield) [inline src/components/EventCard.jsx]: Thinking. Is "local" the browsers local time? I think for Seattle we should just always choose PST.

File: src/components/VotingTable.jsx [hunk 5]
```diff
@@ -98,7 +98,7 @@ const VotingTable = ({ votingData, isPerson }) => (
                     <Link to={`/events/${votingDatum.eventId}`}>
                       <span>{votingDatum.body_name}</span>
                     </Link>
-                    <div>{moment(votingDatum.eventDate, "MM-DD-YYYY HH:mm:ss").format("LLL")}</div>
+                    <div>{moment.utc(votingDatum.eventDate).format("LLL")}</div>
```
Comment (evamaxfield) [inline src/components/VotingTable.jsx]: Same time thing here. PST over browser local.

**Review by evamaxfield — APPROVED** [review]
First: Incredible, and thank you for finishing this PR. Hopefully my reasoning for making the object and how the object's API is laid out make sense. Looks great overall, some minor questions / changes.

File: src/components/EventCard.jsx (hunk 4 above)
Comment (tohuynh) [reply src/components/EventCard.jsx]: Yes, format('LLL') is browsers local time.
New commit always display Seattle time now.

**Review by evamaxfield — APPROVED** [review]

**Pull request merged**